extensions/Qdrant/Qdrant/Internals/Http/SearchVectorsRequest.cs

```diff
@@ -87,9 +88,13 @@ public SearchVectorsRequest HavingSomeTags(IEnumerable<IEnumerable<string>?>? ta
             if (tags == null) { continue; }
 
             var andFilter = new Filter.AndClause();
-            foreach (var tag in tags)
+            foreach (var tag in tags.Where(t => !string.IsNullOrEmpty(t)))
             {
-                if (!string.IsNullOrEmpty(tag))
+                if (tag[0] == '!')
```

Comment:
a tag name could start with `!`, it's a valid char. If one ever uses that, the filtering logic would be incorrect

Reply:
Never used QDrant, thought that ! could be used to indicate that we have a NOT filter in the query. Fixed introducing an internal class tha can represent the filter instead of using the first character with ! to indicate a NOT filter.